inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp

```diff
@@ -0,0 +1,2250 @@
+        L(main_loop_end_label);
+
+        step = 1;
+        L(tail_loop_label);
+        {
+            cmp(reg_work_amount, 1);
+            jl(tail_loop_end_label, T_NEAR);
+
+            // still use xmm on avx2/avx512
+            mov(reg_src_aux1, reg_src);
+            mov(reg_index_oc, dword[reg_index]);
+            add(reg_src_aux1, reg_index_oc);
+            load_scalar(xmm_valTL, ptr[reg_src_aux1], jcp_.src_dt);
+
+            mov(reg_src_aux1, reg_src);
+            mov(reg_index_oc, dword[reg_index + index_stride]);
+            add(reg_src_aux1, reg_index_oc);
+            load_scalar(xmm_valTR, ptr[reg_src_aux1], jcp_.src_dt);
+
+            mov(reg_src_aux1, reg_src);
+            mov(reg_index_oc, dword[reg_index + 2 * index_stride]);
+            add(reg_src_aux1, reg_index_oc);
+            load_scalar(xmm_valBL, ptr[reg_src_aux1], jcp_.src_dt);
+
+            mov(reg_src_aux1, reg_src);
+            mov(reg_index_oc, dword[reg_index + 3 * index_stride]);
+            add(reg_src_aux1, reg_index_oc);
+            load_scalar(xmm_valBR, ptr[reg_src_aux1], jcp_.src_dt);
+
+            load_scalar(xmm_weightL, ptr[reg_src_aux], memory::f32);
+            load_scalar(xmm_weightR, ptr[reg_src_aux + weight_stride], memory::f32);
+            load_scalar(xmm_weightT, ptr[reg_src_aux + 2 * weight_stride], memory::f32);
+            load_scalar(xmm_weightB, ptr[reg_src_aux + 3 * weight_stride], memory::f32);
+
+            uni_vmulps(xmm_valTR, xmm_valTR, xmm_weightR);
+            uni_vmulps(xmm_valBR, xmm_valBR, xmm_weightR);
+            uni_vfmadd231ps(xmm_valTR, xmm_valTL, xmm_weightL);
+            uni_vfmadd231ps(xmm_valBR, xmm_valBL, xmm_weightL);
+            uni_vmulps(xmm_valTR, xmm_valTR, xmm_weightT);
+            uni_vfmadd231ps(xmm_valTR, xmm_valBR, xmm_weightB);
+
+            if (attr_.post_ops_.len_ != 0) {
+                apply_post_ops(jcp_.dst_dt, true);  // process on vmm_val, vmm_val is vmm_valTR, and bc
+            }
+            store_scalar(ptr[reg_dst], xmm_valTR, jcp_.dst_dt);
+
+            add(reg_dst, step * jcp_.dst_data_size);
+            add(reg_src_aux, step * sizeof(float));
+            add(reg_index, step * jcp_.indices_size);
+            sub(reg_work_amount, step);
+
+            jmp(tail_loop_label, T_NEAR);
+        }
+        L(tail_loop_end_label);
+    }
+
+    inline void load_vector(Vmm vmm_src, const Xbyak::Address &op, memory::data_type src_dt) {
+        switch (src_dt) {
+            case memory::f32:
+            case memory::s32:
+                uni_vmovups(vmm_src, op);
+                break;
+            case memory::s8:
+                uni_vpmovsxbd(vmm_src, op);
+                break;
+            case memory::u8:
+                uni_vpmovzxbd(vmm_src, op);
+                break;
+            case memory::bf16:
+                if (isa != cpu::sse42) {
+                    vpmovzxwd(vmm_src, op);
+                } else {
+                    pmovzxwd(vmm_src, op);
```

Comment:
else branch seems useless as well

Reply:
yes, else branch never execute and removed. Thank you!